pyvista/core/filters/data_set.py

```diff
@@ -2511,6 +2511,14 @@ def glyph(
             ``indices``. The values of the range (see ``rng``) affect lookup
             in the table.
 
+            .. note::
+
+                The provided geometry should initially be "pointing" in the
+                positive ``+x`` direction. E.g. use ``direction=(1, 0, 0)``
+                when using some :ref:`Geometric Objects <geometry_api>` or
+                rotate the geometry manually, e.g. with
+                :func:`~pyvista.DataSetFilters.rotate_vector`.
```

Comment:
From https://github.com/pyvista/pyvista/pull/6829#discussion_r1854511065
```suggestion
                The reference direction is relative to ``(1, 0, 0)`` on the
                provided geometry. That is, the provided geometry will be rotated
                from ``(1, 0, 0)`` to the direction of the ``orient`` vector at
                each point.
```

Reply:
The new text is a better, more accurate description. I couldn't find good wording to also add the note for direction objects. So, I think the changes arei good as suggested.

Leaving this open for @MatthewFlamm to commit for co-author credit